tests/test_activity_worker.py

```diff
@@ -87,12 +87,15 @@ def test_activity_worker_gets_tasks(activity_worker_kwargs, tasks):
     Outgoing Command: Assert sent message.
     """
     mock_client = activity_worker_kwargs["client"]
-    mock_client.get_activity_task.side_effect = tasks
+    mock_client.get_activity_task.side_effect = [
+        *tasks, KeyboardInterrupt("This is a keyboard interrupt")
+    ]
 
     worker = ActivityWorker(**activity_worker_kwargs)
-    worker()
+    worker.listen()
```

Comment:
Can we revert these to just testing the basic "process a single task unit" and do a separate test for listen making sure the thing happens?

Reply:
done